src/frontend/js/data/fetchApi.ts

```diff
@@ -0,0 +1,70 @@
+import { appFrontendContext } from './frontEndData';
+import {
+  Actions,
+  User,
+  Role,
+} from './../components/DashboardModerators/common';
+import { handle } from '../utils/errors/handle';
+
+/**
+ * Request API to fetch list data
+ */
+export const fetchListUsers = async (role: Role) => {
+  let response: Response;
+  try {
+    response = await fetch(`/api/v1/users/?role=` + role, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+  } catch (error) {
+    return handle(error);
+  }
+  if (!response.ok) {
+    return handle(
+      new Error(`Failed to load  ${role} list: ${response.status}.`),
+    );
+  }
+  try {
+    return await response.json();
+  } catch (error) {
+    handle(new Error(`Failed to decode JSON in list ${role}: ${error}`));
+  }
+};
+
+/**
+ * Request API to add group moderator for this student
+ */
+export const updateStudentToModerator = async (user: User, action: Actions) => {
+  let response: Response;
+  // Depending on action submit the requested role
+  switch (action) {
+    case Actions.PROMOTE:
+      user.role = Role.MODERATOR;
+      break;
+    case Actions.REVOKE:
+      user.role = Role.STUDENT;
+  }
+  try {
+    response = await fetch(`/api/v1/users/${user.id}/`, {
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRFTOKEN': appFrontendContext.csrftoken,
+      },
+      body: JSON.stringify(user),
+      method: 'PUT',
+    });
+  } catch (error) {
+    return handle(error);
```

Comment:
If it fails, this function silently fail. You are handling the error but the caller never know the api call is failing.

Reply:
Do you have a suggestion? I didn't know what to do with it

Reply:
I suggest you to just throw the error. And it's the caller job to handle it. Maybe it wants to `report` the error ? Do an alternative call, etc.